sdk/servicebus/service-bus/src/connectionContext.ts

```diff
@@ -64,6 +119,46 @@ export namespace ConnectionContext {
     const connectionContext = ConnectionContextBase.create(parameters) as ConnectionContext;
     connectionContext.clientContexts = {};
 
+    let waitForDisconnectResolve: () => void;
+    let waitForDisconnectPromise: Promise<void> | undefined;
+    Object.assign<ConnectionContext, ConnectionContextMethods>(connectionContext, {
+      isConnectionClosing() {
+        // When the connection is not open, but the remote end is open,
+        // then the rhea connection is in the process of terminating.
+        return Boolean(!this.connection.isOpen() && this.connection.isRemoteOpen());
+      },
+      async readyToOpenLink() {
```

Comment:
Let's add a log statement in the beginning and end of this function code. Just in case we are stuck awaiting on something that is not getting resolved. Something like `Waiting to check if we are ready to open link` and then `Ready to open link`

Reply:
Updated!